tests/autorestart_test/test_container_autorestart.py

```diff
@@ -0,0 +1,327 @@
+              can be stopped and then restarted correctly 
+    """
+    kill_process_by_pid(duthost, container_name, program_name, program_status, program_pid)
+
+    logger.info("Waiting until {} container is stopped...".format(container_name))
+    pytest_assert(wait_until(CONTAINER_STOP_THRESHOLD_SECS, 
+                      CONTAINER_CHECK_INTERVAL_SECS,
+                      check_container_status, duthost, container_name, True), 
+                  "Failed to stop {} container".format(container_name))
+    logger.info("{} container is stopped".format(container_name))
+
+    logger.info("Waiting until {} container is restarted...".format(container_name))
+    restarted = wait_until(CONTAINER_RESTART_THRESHOLD_SECS, 
+                  CONTAINER_CHECK_INTERVAL_SECS,
+                  check_container_status, duthost, container_name, False)
+    if not restarted:
+        if is_hiting_start_limit(duthost, container_name):
+            logger.info("{} hits start limit and clear reset-failed flag".format(container_name))
+            duthost.shell("sudo systemctl reset-failed {}.service".format(container_name))
+            duthost.shell("sudo systemctl start {}.service".format(container_name))
+            pytest_assert(wait_until(CONTAINER_RESTART_THRESHOLD_SECS, 
+                      CONTAINER_CHECK_INTERVAL_SECS,
+                      check_container_status, duthost, container_name, False), 
+                  "Failed to restart {} after reset-failed is cleared".format(container_name))
+        else:
+            pytest_assert(False, "Failed to restart {} container".format(container_name)) 
+
+    logger.info("{} container is restarted".format(container_name))
+
+def verify_no_autorestart_with_non_critical_process(duthost, container_name, program_name,
+                                                    program_status, program_pid):
+    """
+    @summary: Killing a non-critical process in a container to verify whether the container 
+              is still in the running state 
+    """
+    kill_process_by_pid(duthost, container_name, program_name, program_status, program_pid)
+
+    logger.info("Checking whether the {} container is still running...".format(container_name))
+    pytest_assert(wait_until(CONTAINER_STOP_THRESHOLD_SECS, 
+                      CONTAINER_CHECK_INTERVAL_SECS,
+                      check_container_status, duthost, container_name, False),
+                  "{} container is stopped unexpectedly".format(container_name))
+    logger.info("{} container is running".format(container_name))
+
+def test_containers_autorestart(duthost):
+    """
+    @summary: Test the auto-restart feature of each contianer against two scenarios: killing
+              a non-critical process to verity the container is still running; killing each
+              critical process to verify the container will be stopped and restarted
+    """ 
+    autorestart_containers = get_autorestart_container_and_state(duthost)
```

Comment:
This variable name is vague and confusing. It should contain mention of "initial state"